cmake/Hipify.cmake

```diff
@@ -1,11 +1,54 @@
 # cmake file to trigger hipify
 
+set(DICT_FILE ${CMAKE_BINARY_DIR}/hipify_output_dict_dump.txt)
+
 # Get the hipify_cli.py source directory from current directory by going to parent directory
-get_filename_component(HIPIFY_SCRIPTS_DIR ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
+get_filename_component(HIPIFY_DIR ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
```

Comment:
Apparently, `CMAKE_CURRENT_LIST_DIR` is only available from CMake version 2.8.3 onwards (https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/Useful-Variables). We should add a minimum cmake version requirement)

Reply:
added the cmake_minimum version statement, kept 3.0 since few of the file() operations require it.